### DISCUSSION TITLE
How do I host glTF files from Amazon S3 and use them with <model-viewer>?

### DISCUSSION THREAD

**Discussion by IvanNov7** [google/model-viewer#1884] [Q&A]
Hello every one!
Trying to run the script with model from amazon s3 but cant load model.
Maybe anyone could tell me where is the problem and what i am doing wrong ?
Here is my example: 
```
<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
<model-viewer src="https://in3dreal.s3.eu-central-1.amazonaws.com/scene.glb" auto-rotate camera-controls></model-viewer>
```

**Comment by cdata** [comment]
Hey @IvanNov7 it looks like you're still not sending CORS headers in the response. Here is a help article I found on Google that describes how to enable CORS for an Amazon S3 bucket: https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html#how-do-i-enable-cors

I'm not an AWS expert, so that's the best information I can offer.

Note that you will need to enable CORS in the server that hosts your models *if and only if* that server is different from the one that hosts the `<model-viewer>` HTML. If you serve them from the same origin (that is, same protocol, domain name and port), then CORS should not be required.

**Comment by raxx21** [comment]
Have you got any solution for this issue ?

**Comment by WadeMorrison** [comment]
Here is what I have used in the "CORS" section under "Permissions" on an S3 bucket. Hope this helps!

`[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]`

**Reply by darinlarimore** [reply]
Thank you this worked for me! @WadeMorrison

**Comment by DanielBustillos** [comment]
The responde of @WadeMorrison helped me. I was struggling with this, I suggest adding this in the documentation as S3 is used for storing this kind of files.

**Reply by elalish** [reply]
Done: https://modelviewer.dev/docs/faq.html#entrydocs-technical-questions-cors
